libsel4/sel4_arch_include/aarch64/sel4/sel4_arch/constants.h

```diff
@@ -52,6 +54,15 @@ typedef enum {
     SEL4_FORCE_LONG_ENUM(seL4_VMFault_Msg),
 } seL4_VMFault_Msg;
 
+/* format of a CHERI fault message */
+typedef enum {
+    seL4_CHERIFault_IP,
+    seL4_CHERIFault_Addr,
+    seL4_CHERIFault_FSR,
+    seL4_CHERIFault_Length,
+    SEL4_FORCE_LONG_ENUM(seL4_CHERIFault_Msg),
+} seL4_CHERIFault_Msg;
```

Comment:
This is a subset of `seL4_VMFault_Msg`, please just use `seL4_VMFault_Msg` instead of creating something almost the same.

Reply:
This is just a place holder (and isn't currently used). There's another `cheriTODO` (also for CHERI-RISC-V) to further modify this message format and send to the user. Also, theoretically speaking, CHERI faults aren't VM faults. I will add another `cheriTODO` comment above this struct.

Reply:
I would say that theoretically speaking, CHERI fault _are_ VM faults, only detected via a different mechanism.

Reply:
"VM" entails an MMU. CHERI faults are merely "hardware capability protection faults." CHERI doesn't necessarily need an MMU or virtualisation (eg VM) enabled and there are already implementations such as CheriFreeRTOS and CHERIoT without virtualisation at all, yet the same types of faults.

Reply:
For reference:
![Screenshot 2024-09-24 at 19 31 25](https://github.com/user-attachments/assets/bde932db-040f-4311-9b70-9ab2443cb71e)

Reply:
I agree that practically CHERI exceptions are different than normal VM faults, so I understand if you add new code for that.